### PR TITLE
Fix build on OpenWrt

### PR DESCRIPTION
**Pull request by neheb**
These are a few patches that are kept in OpenWrt in order to fix compilation with musl. These should be the non-OpenWrt specific ones.

### CODE REVIEW

File: modules/pam_exec/pam_exec.c [hunk 1]
```diff
@@ -180,12 +180,12 @@ call_exec (const char *pam_type, pam_handle_t *pamh,
...
 		{
 		  pam_set_item (pamh, PAM_AUTHTOK, resp);
-		  authtok = strndupa (resp, PAM_MAX_RESP_SIZE);
+		  strcpy (authtok, resp);
 		  _pam_drop (resp);
 		}
 	    }
 	  else
-	    authtok = strndupa (void_pass, PAM_MAX_RESP_SIZE);
+	    strcpy (authtok, void_pass);
```
Comment (t8m) [inline modules/pam_exec/pam_exec.c]: This is unsafe. Although the password if obtained from conversation function cannot be larger than PAM_MAX_RESP_SIZE, in theory some module can set the PAM_AUTHTOK item to any arbitrarily large string. So to be on the safe side, please use strncpy correctly and do it in the above call as well.

**Comment by neheb** [comment]
I think this is what you meant?

File: modules/pam_exec/pam_exec.c [hunk 2]
```diff
@@ -180,12 +180,12 @@ call_exec (const char *pam_type, pam_handle_t *pamh,
 	      if (resp)
 		{
 		  pam_set_item (pamh, PAM_AUTHTOK, resp);
-		  authtok = strndupa (resp, PAM_MAX_RESP_SIZE);
+		  strncpy (authtok, resp, PAM_MAX_RESP_SIZE - 1);
```
Comment (t8m) [inline modules/pam_exec/pam_exec.c]: It would be better to use sizeof(authtok) - 1 instead.

**Comment by neheb** [comment]
done.

**Comment by t8m** [comment]
There were bugs in both Makefile.am and configure.ac patches. I've fixed them.

**Pull request merged**